pkg/server/server.go

```diff
@@ -2443,10 +2443,14 @@ func (s *Server) Decommission(
 
 	for _, nodeID := range nodeIDs {
 		statusChanged, err := s.nodeLiveness.SetMembershipStatus(ctx, nodeID, targetStatus)
-		if err != nil {
-			if errors.Is(err, liveness.ErrMissingRecord) {
+		if errors.Is(err, liveness.ErrMissingRecord) {
+			// If the node is unknown (no liveness record) and we're trying to
+			// decommission, we go ahead with cleanup anyway. In particular, we want
+			// to remove stray node status entries even if no liveness entry exists.
+			if !targetStatus.Decommissioning() && !targetStatus.Decommissioned() {
```

Comment:
can we make this code less tangly and handle `ErrMissingLivenessRecord` right here? If we get here and we are really just trying to nuke the node, invoke the cleanup code like below (maybe after extraction into a method). I just don't want to consider this case in the rest of the method and I don't think we should be going through the event log, etc.

Reply:
Sure, I'll pull the cleanup out into a separate method and return early.